src/main/java/com/google/cloud/spanner/pgadapter/wireprotocol/CopyDoneMessage.java

```diff
@@ -51,13 +51,15 @@ protected void sendPayload() throws Exception {
         // Spanner returned an error when trying to commit the batch of mutations.
         mw.writeMutationsToErrorFile();
         mw.closeErrorFile();
+        this.connection.setStatus(ConnectionStatus.IDLE);
         this.connection.removeActiveStatement(this.statement);
         throw e;
       }
     } else {
       mw.closeErrorFile();
     }
     new ReadyResponse(this.outputStream, ReadyResponse.Status.IDLE).send();
+    this.connection.setStatus(ConnectionStatus.IDLE);
```

Comment:
I think that this call to `setStatus(..)` and also the above one are causing a build error, because `ConnectionStatus` is not imported in this file. Would you mind taking a look at that?

Reply:
Fixed now.  Originally this PR was supposed to require changes from https://github.com/GoogleCloudPlatform/pgadapter/pull/40